components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Discovery/src/Services/DiscoveryServices.cs

```diff
@@ -437,8 +438,11 @@ public class DiscoveryServices : IDiscoveryServices, IScannerServices, IDisposab
         private Task<(IPEndPoint, Uri)> GetHostEntryAsync(
             Uri discoveryUrl) {
             return Try.Async(async () => {
+
                 var entry = await Dns.GetHostEntryAsync(discoveryUrl.DnsSafeHost);
-                foreach (var address in entry.AddressList) {
+                // only pick-up the IPV4 addresses
+                foreach (var address in entry.AddressList
```

Comment:
Hmm.. algorithm should be to send ping to every address then wait for the first to return and cancel outstanding requests? Any way hosts don't have to answer ping. But I don't have all the context here...

Reply:
This is to pick an endpoint that can be connected to using the discovery service client.  Ipv4 must be preferred, however - we should make sure that there is at least an item in the list, e.g. if there are no ipv4 addresses we should still use ipv6.